cmd/termui/provider/logProvider.go

```diff
@@ -21,22 +22,25 @@ const (
 )
 
 // InitLogHandler will open the websocket and set the log level
-func InitLogHandler(nodeURL string, profile logger.Profile, useWss bool) error {
+func InitLogHandler(presenter PresenterHandler, nodeURL string, profile logger.Profile, useWss bool) error {
 	var err error
 	scheme := ws
 	if useWss {
 		scheme = wss
 	}
-	for {
-		webSocket, err = openWebSocket(scheme, nodeURL, profile)
-		if err == nil {
-			break
-		}
-		log.Error("init logger", "error", err)
+	go func() {
+		for {
+			webSocket, err = openWebSocket(scheme, nodeURL, profile)
+			if err != nil {
+				_, _ = presenter.Write([]byte(fmt.Sprintf("termui websocket error, retrying in %v...", retryDuration)))
```

Comment:
check.IfNil for presented. Add context for infinite goRoutine in order to have Close() as well.

Reply:
this is a separate application. The goroutine is stopped when closing the application. 
Regarding the check.IfNil, the presenter is initialized before sending it to the method so I don't think there is a need to check it for nil because if it would be nil, the application wouldn't arrive here